react/week3/todolist/src/TodoList.js

```diff
@@ -0,0 +1,114 @@
+import React, { useEffect, useState } from "react";
+import DatePicker from "react-datepicker";
+import { toDate } from "date-fns";
+import "react-datepicker/dist/react-datepicker.css";
+import TodoItem from "./TodoItem";
+const api_url =
+  "https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw";
+
+function TodoList() {
+  const [itemsArray, setItemsArray] = useState([]);
+  const [newDescription, setNewDescription] = useState("");
+  const [newDeadline, setNewDeadline] = useState(null);
+
+  const fetchFromApi = () => {
+    fetch(api_url)
+      .then((response) => response.json())
+      .then((data) => {
+        setItemsArray(data);
+      });
+  };
+
+  useEffect(() => {
+    fetchFromApi();
+  }, []);
+
+  const addNewDescription = (event) => {
+    setNewDescription(event.target.value);
+  };
+
+  const addNewDeadline = (input) => {
+    setNewDeadline(toDate(input));
+  };
+
+  const addNewItem = (e) => {
+    e.preventDefault();
+    const newDeadlineMonth =
+      newDeadline.getMonth() + 1 < 10
+        ? `0${newDeadline.getMonth() + 1}`
+        : newDeadline.getMonth() + 1;
+    const newDeadlineDate =
+      newDeadline.getDate() < 10
+        ? `0${newDeadline.getDate()}`
+        : newDeadline.getDate();
```

Comment:
I'd take a look at this method: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart